{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "jewish-vinyl",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Attaching SeuratObject\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘SummarizedExperiment’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SeuratObject’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Seurat’:\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘SingleR’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:celldex’:\n",
      "\n",
      "    BlueprintEncodeData, DatabaseImmuneCellExpressionData,\n",
      "    HumanPrimaryCellAtlasData, ImmGenData, MonacoImmuneData,\n",
      "    MouseRNAseqData, NovershternHematopoieticData\n",
      "\n",
      "\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "\n",
      "Attaching package: ‘AnnotationDbi’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "An object of class Seurat \n",
       "20570 features across 18748 samples within 2 assays \n",
       "Active assay: integrated (2000 features, 2000 variable features)\n",
       " 1 other assay present: RNA\n",
       " 2 dimensional reductions calculated: pca, umap"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“'ImmGenData' is deprecated.\n",
      "Use 'celldex::ImmGenData' instead.\n",
      "See help(\"Deprecated\")”\n",
      "using temporary cache /tmp/RtmpM9NoIh/BiocFileCache\n",
      "\n",
      "snapshotDate(): 2020-10-27\n",
      "\n",
      "see ?celldex and browseVignettes('celldex') for documentation\n",
      "\n",
      "downloading 1 resources\n",
      "\n",
      "retrieving 1 resource\n",
      "\n",
      "loading from cache\n",
      "\n",
      "see ?celldex and browseVignettes('celldex') for documentation\n",
      "\n",
      "downloading 1 resources\n",
      "\n",
      "retrieving 1 resource\n",
      "\n",
      "loading from cache\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "class: SummarizedExperiment \n",
       "dim: 22134 830 \n",
       "metadata(0):\n",
       "assays(1): logcounts\n",
       "rownames(22134): Zglp1 Vmn2r65 ... Tiparp Kdm1a\n",
       "rowData names(0):\n",
       "colnames(830):\n",
       "  GSM1136119_EA07068_260297_MOGENE-1_0-ST-V1_MF.11C-11B+.LU_1.CEL\n",
       "  GSM1136120_EA07068_260298_MOGENE-1_0-ST-V1_MF.11C-11B+.LU_2.CEL ...\n",
       "  GSM920654_EA07068_201214_MOGENE-1_0-ST-V1_TGD.VG4+24ALO.E17.TH_1.CEL\n",
       "  GSM920655_EA07068_201215_MOGENE-1_0-ST-V1_TGD.VG4+24ALO.E17.TH_2.CEL\n",
       "colData names(3): label.main label.fine label.ont"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following arguments are not used: drop”\n"
     ]
    }
   ],
   "source": [
    "library(dplyr)\n",
    "library(Seurat)\n",
    "library(patchwork)\n",
    "library(SoupX)\n",
    "library(ggplot2)\n",
    "library(celldex)\n",
    "library(SingleR)\n",
    "library(org.Mm.eg.db)\n",
    "library(SingleCellExperiment)\n",
    "\n",
    "# 3. Annotation using SingleR package -----------------------------------------------------------------------------\n",
    "\n",
    "integrated_annotation <- readRDS(file = \"/home/phil/integrated_brucei/clustering_integrated.rds\")\n",
    "integrated_annotation\n",
    "\n",
    "# load ImmGenData library\n",
    "ref <- ImmGenData()\n",
    "ref\n",
    "\n",
    "# annotate integrated_annotation variable\n",
    "whole_annotation <- SingleR(test = SummarizedExperiment(integrated_annotation), ref = ref, assay.type.test=1, \n",
    "                            labels = ref$label.main)\n",
    "whole_annotation\n",
    "head(integrated_annotation@meta.data)\n",
    "\n",
    "# Copy over the labels and pruned.labels (Note: any other column of the results could be used as well)\n",
    "integrated_annotation$SingleR.pruned.calls <- whole_annotation$pruned.labels\n",
    "integrated_annotation$SingleR.calls <- whole_annotation$labels\n",
    "\n",
    "#Run UMAP------------------------------------------------------------------------------------------\n",
    "integrated_annotation <- RunUMAP(integrated_annotation, dims = 1:10)\n",
    "DimPlot(integrated_annotation, reduction = \"umap\", group.by = \"SingleR.calls\", label = TRUE, pt.size = 0.5)\n",
    "head(integrated_annotation@meta.data)\n",
    "integrated_annotation\n",
    "\n",
    "#Annotation Diagnostics----------------------------------------------------------------------------\n",
    "pred.grun <- SingleR(test=as.SingleCellExperiment(integrated_annotation), ref=ref, labels=ref$label.main, de.method=\"wilcox\")\n",
    "table(pred.grun$labels)\n",
    "plotScoreHeatmap(pred.grun)\n",
    "plotDeltaDistribution(pred.grun, ncol = 3)\n",
    "\n",
    "all.markers <- metadata(pred.grun)$de.genes\n",
    "sceG$labels <- pred.grun$labels\n",
    "\n",
    "\n",
    "# Next, switch the identity class of all cells to reflect replicate ID\n",
    "Idents(integrated_annotation) <- \"SingleR.calls\"\n",
    "\n",
    "# How many cells are in each cluster\n",
    "table(Idents(integrated_annotation))\n",
    "\n",
    "# How many cells are in each cluster\n",
    "integrated_annotation_counts <- table(Idents(integrated_annotation))\n",
    "head(integrated_annotation_counts)\n",
    "\n",
    "integrated_annotation_long <- as.data.frame(integrated_annotation_counts)\n",
    "head(integrated_annotation_long)\n",
    "colnames(integrated_annotation_long) <- c(\"cell_type\", \"Cell_count\")\n",
    "head(integrated_annotation_long)\n",
    "rownames(integrated_annotation_long) <- integrated_annotation_long$x\n",
    "\n",
    "integrated_annotation_long\n",
    "\n",
    "# Create ggplot2 plot scaled to 1.00\n",
    "bar_counts <- ggplot(integrated_annotation_long,            \n",
    "              aes(x = cell_type, y = Cell_count,\n",
    "                  fill = cell_type)) +\n",
    "  geom_bar(stat = \"identity\", color =\"black\") + \n",
    "  #scale_y_continuous(labels = scales::percent_format()) + \n",
    "  scale_x_discrete(name =\"Cell type\") +\n",
    "  theme(legend.position=\"right\") +\n",
    "  theme(axis.text.x=element_blank(), axis.ticks.x=element_blank())\n",
    "\n",
    "# Draw ggplot2 plot scaled to 1.00\n",
    "bar_counts                               \n",
    "\n",
    "# save annotated file\n",
    "saveRDS(integrated_annotation, file = \"annotation_integrated.rds\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
